src/main/java/net/minecraftforge/event/world/DimensionPreloadEvent.java

```diff
@@ -0,0 +1,35 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+/**
+ * Created by goreacraft on 17/03/03.
+ *
+ * This event is fired when DimensionManager is starting to load a new dimmension
+ * dimension 0 (Overworld) will not trigger this event
+ * {@link DimensionManager#initDimension(int)},
+ * <br>
+ * This event is {@link Cancelable}.<br>
```

Comment:
i will give soon an example of use in more detail with explanation, i hope by tomorrow...
for now you can see what the forgeessentials guys had to do using mixin to cancel function initDimension and make their multiverse module work. Having this event it will not be necessary to use mixin. 
I am working on a multiverse module myself and i "hit this wall" but not having an event for this, so i fix it by adding it.
I am sorry if i don't know how to make my intentions more clear then this, if you still need more clarifications i will try with the example soon.